tests/vector/vecas.cpp

```diff
@@ -0,0 +1,31 @@
+/* RUN: %{execute}%s
+
+   Test some vec<> as method behaviour
+*/
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <boost/test/minimal.hpp>
+
+using namespace cl::sycl;
+
+int test_main(int argc, char *argv[]) {
+	vec<int, 4> v = {1, 2, 3, 4};
+
+	BOOST_CHECK(v.x() == 1);
+	BOOST_CHECK(v.y() == 2);
+	BOOST_CHECK(v.z() == 3);
+	BOOST_CHECK(v.w() == 4);
+
+	vec<unsigned int, 4> v2 = v.as<cl::sycl::vec<unsigned int, 4>>();
+	BOOST_CHECK(v2.x() == 1);
+	BOOST_CHECK(v2.y() == 2);
+	BOOST_CHECK(v2.z() == 3);
+	BOOST_CHECK(v2.w() == 4);
+
+	vec<unsigned char, 16> v3 = v.as<cl::sycl::vec<unsigned char, 16>>();
+	BOOST_CHECK(v3[0] == 1);
+	BOOST_CHECK(v3[4] == 2);
+	BOOST_CHECK(v3[8] == 3);
+	BOOST_CHECK(v3[12] == 4);
+	return 0;
+}
```

Comment:
Probably we need some `asT` different from `T` to be more exhaustive.

Reply:
char, 16 is different to int, 4, I've added a float/int test as well in next push.

Reply:
Sorry I was completely wrong about the code. You are right.

But then the code could be simplified with some `auto` everywhere
```
auto v3 = v.as<cl::sycl::vec<unsigned char, 16>>();
```